feat: Mission Control docs (#3916)
langchain-ai/docs: src/langsmith/self-hosted-mission-control.mdx

---
title: Install Mission Control
sidebarTitle: Mission Control
description: Install Mission Control, an in-cluster console for monitoring, configuring, and operating self-hosted LangSmith on Kubernetes.
---

Mission Control is an in-cluster console for monitoring, configuring, and operating LangSmith on Kubernetes. It runs inside your cluster and is accessed with `kubectl port-forward` by default, so no ingress is required.

> This guide has two install paths:

Reviewer: ```suggestion
There are two install paths:
```

| Path | Best for |
|------|----------|
| [Quick install](#quick-install) | Customers who can run a reviewed shell installer and want the shortest setup. |
| [Manual install](#manual-install) | Organizations that do not allow installer scripts or need each Kubernetes command reviewed. |

The public install assets are:

- `install-script.sh`: one installer with separate `prereqs`, `namespace`, `secret`, `values`, `install`, and `forward` steps.
- `values.yaml`: default Helm values for a port-forward-only install.

> Mission Control images are published in Docker Hub at `langchain/langsmith-mission-control`. The latest images can be checked with:

Reviewer: ```suggestion
Mission Control images are published in Docker Hub at `langchain/langsmith-mission-control`. Check the latest images with:
```

./install-script.sh all -f values.yaml
```

The `all` step:

- Checks required tools and RBAC.
- Creates the `langsmith` namespace.
- Prompts for a Mission Control username and password.
- Stores those credentials in the `mission-control-auth` Kubernetes Secret.
- Writes `values.yaml` if one does not already exist.
- Installs from the public Helm chart repository if you are not running from a local chart checkout.
- Installs Mission Control with Helm.

The RBAC check runs:

```bash
kubectl auth can-i create clusterrole
kubectl auth can-i create clusterrolebinding
kubectl auth can-i create serviceaccount -n langsmith
kubectl auth can-i create deployment -n langsmith
kubectl auth can-i create secret -n langsmith
```

If your organization intentionally blocks `kubectl auth can-i` but Helm installs are approved through another control path, run:

```bash
./install-script.sh all -f values.yaml --skip-rbac-check
```

### Access the UI

After the install finishes, start a local port-forward:

```bash
./install-script.sh forward
```

Open [http://localhost:3000](http://localhost:3000/) and log in with the username and password you entered.

### Review the script first

> The Quick install path downloads the script before running it, so you can review `install-script.sh` locally before the third command.

Reviewer: ```suggestion
The quick install path downloads the script before running it, so you can review `install-script.sh` locally before the third command.
```

### Edit values before install

> The Quick install path also downloads `values.yaml` before running the installer. Review or edit that file before the third command when you need to change namespace, resources, ingress, feature flags, or diagnostic persistence.

Reviewer: ```suggestion
The quick install path also downloads `values.yaml` before running the installer. Review or edit that file before the third command if you need to change namespace, resources, ingress, feature flags, or diagnostic persistence.
```

  </Step>
</Steps>

## Upgrade

Download the latest public values file, merge in any local changes you need, then run:

```bash
helm repo update langchain

helm upgrade --install mission-control langchain/langsmith-mission-control \
  --namespace langsmith \
  --values values.yaml \
  --rollback-on-failure
```

If you are working from a local chart checkout instead:

```bash
helm upgrade --install mission-control . \
  --namespace langsmith \
  --values values.yaml \
  --rollback-on-failure
```

If you installed with the quick script and kept it locally:

```bash
./install-script.sh install -f values.yaml
```

## Uninstall

```bash
helm uninstall mission-control -n langsmith
```

This removes the Mission Control release. It does not delete your namespace or unrelated LangSmith resources.

Optional cleanup of Mission Control-owned Secrets:

```bash
kubectl delete secret -n langsmith \
  mission-control-auth \
  mission-control-draft \
  mission-control-deployed \
  mission-control-backup \
  mission-control-history \
  mission-control-alerts-config \
  mission-control-alerts-log \
  mission-control-alerts-key \
  mission-control-setup-token \
  --ignore-not-found
```

> 

Reviewer: To organize the headings more, maybe we should add a `## Additional resources` and make Troubleshooting, Permissions reference, etc. all H3

Author: I've added the feedback you suggested

---
title: Install Mission Control
sidebarTitle: Mission Control
description: Install Mission Control, an in-cluster console for monitoring, configuring, and operating self-hosted LangSmith on Kubernetes.
---

Mission Control is an in-cluster console for monitoring, configuring, and operating LangSmith on Kubernetes. It runs inside your cluster and is accessed with `kubectl port-forward` by default, so no ingress is required.

There are two install paths:

| Path | Best for |
|------|----------|
| [Quick install](#quick-install) | Customers who can run a reviewed shell installer and want the shortest setup. |
| [Manual install](#manual-install) | Organizations that do not allow installer scripts or need each Kubernetes command reviewed. |

The public install assets are:

- `install-script.sh`: one installer with separate `prereqs`, `namespace`, `secret`, `values`, `install`, and `forward` steps.
- `values.yaml`: default Helm values for a port-forward-only install.

> Mission Control images are published to two Docker Hub repositories—`langchain/mission-control-backend` and `langchain/mission-control-frontend`. The latest images can be checked with:

Reviewer: ```suggestion
Mission Control images are published to two Docker Hub repositories: `langchain/mission-control-backend` and `langchain/mission-control-frontend`. The latest images can be checked with:
```

> Mission Control images are published to two Docker Hub repositories — `langchain/mission-control-backend` and `langchain/mission-control-frontend`. The latest images can be checked with:

Reviewer: ```suggestion
Mission Control images are published to two Docker Hub repositories—`langchain/mission-control-backend` and `langchain/mission-control-frontend`. The latest images can be checked with:
```
